test/bg/user-script-registry.test.js

```diff
@@ -31,4 +31,27 @@ describe('bg/user-script-registry', () => {
       });
     });
   });
+
+  it('match pattern respects query string', async () => {
```

Comment:
Instead of `it('match pattern ...`, this should be a test of MatchPattern.doMatch() directly.  That's the code you're changing.  And it would also need less setup code, tested that way.

Reply:
Should I add a `test/third-party` directory and `MatchPattern.test.js` then?

Reply:
Yes